src/service_daemon.rs

```diff
@@ -1914,6 +2073,20 @@ pub enum ServiceEvent {
     SearchStopped(String),
 }
 
+/// All possible events sent to the client from the daemon
+/// regarding host resolution.
+#[derive(Debug)]
+pub enum ResolutionEvent {
+    /// Started searching for the ip address of a hostname.
+    SearchStarted(String),
+    /// One or more addresses for a hostname has been found.
+    HostnameAddressesFound(String, HashSet<IpAddr>),
+    /// One or more addresses for a hostname has been removed.
+    HostnameAddressesRemoved(String, HashSet<IpAddr>),
+    /// Stopped searching for the ip address of a hostname.
+    SearchStopped(String),
+}
```

Comment:
I wonder if `String` (i.e. the hostname) is needed in each variant. If `ResolutionEvent` is only sent to a given channel created by a `resolve_*` call, then the `String` is redundant as the caller has it.  And the variant `HostnameAddressesFound` can be named as `AddressesFound`.

If we keep the `String` value, then a question is:  what happens if `ResolutionEvent` is for things other than `hostname`?   I think it's more clearly to name `SearchStarted` as `HostnameSearchStarted`.

Reply:
> I wonder if `String` (i.e. the hostname) is needed in each variant. If `ResolutionEvent` is only sent to a given channel created by a `resolve_*` call, then the `String` is redundant as the caller has it.

That's right, I hadn't given it much thought. We could certainly do without it. But I see you're returning the `service_type` in all the `ServiceEvent`s as well, not sure if we should keep the same behaviour to make it consistent?

I could also see someone using it to convert several `Receive<Event>` structs into streams, and combine them into one large stream that combines the handling for all channels into a single loop, without having to do reverse lookups for figuring out which hostname the event belongs to (I'm not sure if that even works).

> And the variant `HostnameAddressesFound` can be named as `AddressesFound`.
>
> If we keep the `String` value, then a question is: what happens if `ResolutionEvent` is for things other than hostname? I think it's more clearly to name `SearchStarted` as `HostnameSearchStarted`.

I guess it depends on what you'd want to extend it to do in the future? It might be better to just name the whole enum `HostnameResolutionEvent`, and rather make other `-Event` structs for other kinds of functionality. Referring back to the decision not to merge this hostname resolution functionality with `ServiceEvent`, in order to avoid figuring out which enum variant belonged to what querier, and to avoid having the user having to handle enum variants that will never appear for the type of query they're using.

I think the only argument for keeping the hostname in the `SearchStarted` and `SearchStopped` events would be consistency with `ServiceEvent`.

What do you think?

Reply:
> But I see you're returning the `service_type` in all the `ServiceEvent`s as well, not sure if we should keep the same behaviour to make it consistent?

Good point! I agree it's better to keep the the hostname.

> I guess it depends on what you'd want to extend it to do in the future? It might be better to just name the whole enum HostnameResolutionEvent, and rather make other -Event structs for other kinds of functionality.

I see that `avahi` supports `avahi-resolve-address` , i.e. reverse lookup hostname. I don't think if we will support it in future. And again, I further don't know if it's good idea to extend the same `ResolutionEvent` for that case.

All things considered, I think your idea of just naming it `HostnameResolutionEvent` is good.  And add `#[non_exhaustive]` just in case.